File: src/191.number_of_1_bits.kt
fun main() {
    println(hammingWeight(3));
    println(hammingWeight(11));
    println(hammingWeight(128));
    println(hammingWeight(2147483645));
}

fun hammingWeight(n: Int): Int {
    var result = 0
    var num = n

    while (num > 0) {
        if (num.and(1) > 0) {
            result++
        }

        num = num.shr(1)
    }

    return result
}

File: lib/191.number_of_1_bits.dart
void main() {
  print(hammingWeight(3));
  print(hammingWeight(11));
  print(hammingWeight(128));
  print(hammingWeight(2147483645));
}

int hammingWeight(int n) {
  var result = 0;

  while (n > 0) {
    if (n & 1 > 0) {
      result++;
    }

    n >>= 1;
  }

  return result;
}
